src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -409,12 +427,22 @@ class ProtobufPlugin implements Plugin<Project> {
           }
         }
       } else {
-        project.sourceSets.each { sourceSet ->
-          Task javaCompileTask = project.tasks.getByName(sourceSet.getCompileTaskName("java"))
-          project.protobuf.generateProtoTasks.ofSourceSet(sourceSet.name).each { generateProtoTask ->
-            javaCompileTask.dependsOn(generateProtoTask)
-            generateProtoTask.getAllOutputDirs().each { dir ->
-              javaCompileTask.source project.fileTree([dir:dir])
+        project.sourceSets.each { SourceSet sourceSet ->
```

Comment:
Do you have plan to add support for Android projects?

Reply:
Yes, I have a PR that's ready to be posted after this one is merged.